gsa/src/web/pages/extras/trashcanpage.js

```diff
@@ -119,105 +115,85 @@ const separateByUsageType = inputList => {
   return {scan, compliance};
 };
 
-class Trashcan extends React.Component {
-  constructor(...args) {
-    super(...args);
-    this.state = {
-      trash: undefined,
-      loading: false,
-    };
-
-    this.createContentRow = this.createContentRow.bind(this);
-    this.createContentsTable = this.createContentsTable.bind(this);
-    this.getTrash = this.getTrash.bind(this);
-    this.handleEmpty = this.handleEmpty.bind(this);
-    this.handleDelete = this.handleDelete.bind(this);
-    this.handleRestore = this.handleRestore.bind(this);
-    this.handleErrorClose = this.handleErrorClose.bind(this);
-  }
+const Trashcan = () => {
+  const gmp = useGmp();
+  const [, renewSessionTimeout] = useUserSessionTimeout();
 
-  componentDidMount() {
-    this.getTrash();
-  }
+  const [trash, setTrash] = useState({});
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState();
 
-  getTrash() {
-    const {gmp} = this.props;
+  const getTrash = useCallback(() => {
```

Comment:
Should be renamed to something more meaningful like loadTrashCanData or something like that.